k6-operator: adjust installation instructions (#1981)
grafana/k6-docs: docs/sources/k6/next/set-up/set-up-distributed-k6/install-k6-operator.md

## Deploy the operator

> There are three different methods to deploy the k6 Operator. The first two methods, with bundle and with Helm, install the latest official release of the k6 Operator by default. The third method installs from the branch of the [k6-operator repository](https://github.com/grafana/k6-operator) and is meant for development purposes or for folk who have their own `kustomize` pipeline on top.

Reviewer: ```suggestion
There are three different methods to deploy the k6 Operator. The first two methods, with bundle and Helm, install the latest official release of the k6 Operator by default. The third method installs from the branch of the [k6-operator repository](https://github.com/grafana/k6-operator) and is intended for development purposes or for users who have their own `kustomize` pipeline.
```

### Deploy with Makefile

> This is a more manual, low-level way to install the k6 Operator. It installs from the branch of the k6-operator repository. By default, it's the `main` branch, which is not guaranteed to be always stable. In general case, deployment from the branch is meant for development purposes.

Reviewer: ```suggestion
This method installs the k6 Operator from the GitHub repository. By default, it uses the `main` branch, which is not guaranteed to be stable at all times. This method is helpful for:

- Development purposes.
- Users who have a `kustomize` pipeline and want to adjust manifests with it.
```

Author: :thinking: Strictly speaking, bundle method also installs from the Github repository... It just uses the file that is changing only on the releases and can be considered an artifact. IOW, it seems like there is a bit of a loss of information here. 

What about this version?
```suggestion
This method installs the k6 Operator from the GitHub repository, using the latest files in a branch. By default, it uses the `main` branch, which is not guaranteed to be stable at all times. This method is helpful for:
- Development purposes.
- Users who have a `kustomize` pipeline and want to adjust manifests with it.
```

Reviewer: That looks good to me! 🙇

> However, this method can also be useful for folk who have the `kustomize` pipeline and wish to adjust manifests with `kustomize`. To use this method for regular, production deployments, it is recommended to do that from the tagged commits:

Reviewer: ```suggestion
To use this method for production deployments, it's recommended to do that from the tagged commits:
```

git checkout v0.0.22
```

The tagged commits correspond to the official releases and are expected to be stable.

In order to install the operator with the Makefile, you need:

- [go](https://go.dev/doc/install)
- [kustomize](https://kubectl.docs.kubernetes.io/installation/kustomize/)

Then, you can install the k6 Operator by running the command below from the root of the cloned repository:

```bash
make deploy
```

### Install the CRDs

The k6 Operator includes custom resources called `TestRun` and `PrivateLoadZone`. They're automatically installed when you use one of the three installation methods above. But you can also manually install the CRDs by running:

```bash
make install
```

## Uninstall k6 Operator

Removal of the k6 Operator depends on the installation method. If you use bundle installation, you can remove all of the resources created by the k6 Operator with the following:

```bash
curl https://raw.githubusercontent.com/grafana/k6-operator/main/bundle.yaml | kubectl delete -f -
```

If you use Helm installation, then removal should be done with the `helm` command:

```bash
helm uninstall k6-operator
```

> Finally, if you use Makefile installation, use `make` command to uninstall:

Reviewer: ```suggestion
Finally, if you use Makefile installation, use the `make` command to uninstall:
```
